### ISSUE TITLE
Twitter API being phased out, twitter plugin will be broken

### ISSUE THREAD

**Issue by katanacrimson** [phergie/phergie#86]
Just got an email from Twitter that I think you guys should be aware of.

> Over the coming weeks, we will be making two important updates that will impact how you interact with Twitter applications. We are sending this notice to all Twitter users to make sure you are aware of these changes.
> 
> What are applications?
> 
> There are over 250,000 applications built using the Twitter API. To use most applications, you first authorize the application to access your Twitter account, after which you can use it to read and post Tweets, discover new users and more. Applications come in many varieties, including desktop applications like TweetDeck, Seesmic, or EchoFon, websites such as TweetMeme, fflick, or Topsy, or mobile applications such as Twitter for iPhone, Twitter for Blackberry, or Foursquare.
> 
> Update 1: New authorization rules for applications
> 
> Starting August 31, all applications will be required to use “OAuth” to access your Twitter account.
> 
> What's OAuth?
> 
> OAuth is a technology that enables applications to access Twitter on your behalf with your approval without asking you directly for your password.
> Desktop and mobile applications may still ask for your password once, but after that request, they are required to use OAuth in order to access your timeline or allow you to tweet.
> What does this mean for me?
> 
> Applications are no longer allowed to store your password.
> If you change your password, the applications will continue to work.
> Some applications you have been using may require you to reauthorize them or may stop functioning at the time of this change.
> All applications you have authorized will be listed at http://twitter.com/settings/connections.
> You can revoke access to any application at any time from the list.
> Update 2: t.co URL wrapping
> 
> In the coming weeks, we will be expanding the roll-out of our link wrapping service t.co, which wraps links in Tweets with a new, simplified link. Wrapped links are displayed in a way that is easier to read, with the actual domain and part of the URL showing, so that you know what you are clicking on. When you click on a wrapped link, your request will pass through the Twitter service to check if the destination site is known to contain malware, and we then will forward you on to the destination URL. All of that should happen in an instant.
> 
> You will start seeing these links on certain accounts that have opted-in to the service; we expect to roll this out to all users by the end of the year. When this happens, all links shared on Twitter.com or third-party apps will be wrapped with a t.co URL.
> 
> What does this mean for me?
> 
> A really long link such as http://www.amazon.com/Delivering-Happiness-Profits-Passion-Purpose/dp/0446563048 might be wrapped as http://t.co/DRo0trj for display on SMS, but it could be displayed to web or application users as amazon.com/Delivering- or as the whole URL or page title.
> You will start seeing links in a way that removes the obscurity of shortened links and lets you know where each link will take you.
> When you click on these links from Twitter.com or a Twitter application, Twitter will log that click. We hope to use this data to provide better and more relevant content to you over time.
> Thanks for reading this important update. Come and check what's new at http://twitter.com.
> 
> Thanks,
> The Twitter Team

tl;dr: Twitter API is being phased out, and only the Oauth API will remain.
